backend/canisters/escrow/impl/src/queries/lookup_swap.rs

```diff
@@ -14,7 +14,13 @@ fn lookup_swap_impl(args: Args, state: &RuntimeState) -> Response {
         return SwapNotFound;
     };
 
-    let accepting_principal = args.accepting_principal.unwrap_or(state.env.caller());
+    let caller = state.env.caller();
+
+    if !swap.is_public && !swap.is_admin(caller) {
```

Comment:
I think you should be able to look it up if you're the offerer, the accepter it is restricted to, or an admin.